In [1]:
import keras
from keras.models import Sequential
from keras.layers import Dense
from keras.optimizers import Adam

Using TensorFlow backend.


Drug vaccination between age 13-100 years
people>65 age   observed some side effects(denoted as 1)
people<65 age   no side effects(denoted as 0)

In [2]:
import numpy as np
from random import randint
from sklearn.preprocessing import MinMaxScaler

In [6]:
train_sample=[]
train_label=[]
for i in range(1000):
    younger_ages=randint(13,64)
    train_sample.append(younger_ages)
    train_label.append(0)
    
    older_ages=randint(65,100)
    train_sample.append(older_ages)
    train_label.append(1)

In [7]:
train_sample

[64,
 68,
 20,
 82,
 23,
 83,
 15,
 69,
 58,
 75,
 58,
 81,
 30,
 85,
 38,
 81,
 41,
 92,
 61,
 73,
 14,
 69,
 36,
 99,
 63,
 91,
 45,
 93,
 41,
 72,
 52,
 97,
 51,
 73,
 27,
 90,
 43,
 86,
 14,
 71,
 32,
 94,
 32,
 95,
 41,
 65,
 56,
 89,
 24,
 76,
 33,
 89,
 14,
 82,
 55,
 75,
 29,
 68,
 32,
 68,
 13,
 94,
 31,
 68,
 54,
 83,
 31,
 72,
 34,
 76,
 55,
 91,
 43,
 93,
 48,
 82,
 37,
 79,
 18,
 94,
 16,
 69,
 28,
 77,
 53,
 100,
 37,
 70,
 50,
 74,
 62,
 84,
 19,
 78,
 48,
 91,
 14,
 90,
 14,
 68,
 26,
 87,
 27,
 71,
 32,
 76,
 41,
 99,
 22,
 95,
 47,
 91,
 20,
 74,
 37,
 67,
 52,
 93,
 63,
 95,
 46,
 81,
 56,
 98,
 46,
 92,
 38,
 82,
 18,
 73,
 40,
 69,
 51,
 97,
 33,
 82,
 31,
 95,
 27,
 96,
 20,
 95,
 54,
 93,
 60,
 96,
 52,
 94,
 26,
 85,
 38,
 88,
 45,
 69,
 32,
 91,
 28,
 93,
 61,
 70,
 49,
 98,
 58,
 95,
 58,
 89,
 53,
 92,
 49,
 94,
 32,
 84,
 45,
 90,
 63,
 74,
 17,
 84,
 14,
 79,
 55,
 100,
 42,
 74,
 55,
 76,
 54,
 95,
 56,
 98,
 43,
 94,
 25,
 66,
 24,
 66,
 48,
 73,
 48,
 77

In [8]:
train_label

[0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,


In [9]:
#convert the values into numpy array
train_sample=np.array(train_sample)
train_sample

array([64, 68, 20, ..., 65, 16, 83])

In [10]:
train_label=np.array(train_label)
train_label

array([0, 1, 0, ..., 1, 0, 1])

In [18]:
#scalling the values in between 0 to 1
sc=MinMaxScaler(feature_range=(0,1)) #here we are using feature_range the values we have to convert particular range
sc_train_sample=sc.fit_transform(train_sample.reshape(-1,1))
sc_train_sample

array([[0.5862069 ],
       [0.63218391],
       [0.08045977],
       ...,
       [0.59770115],
       [0.03448276],
       [0.8045977 ]])

In [51]:
model=Sequential()

#Adding the 1st hidden layer and input layer
model.add(Dense(units = 10, kernel_initializer = 'he_normal', activation='relu', input_dim = 1))

#Adding the 2nd hidden layer
model.add(Dense(units = 20, kernel_initializer = 'he_normal', activation='relu'))

#Adding the 3rd hidden layer
model.add(Dense(units = 15, kernel_initializer = 'he_normal', activation='relu'))


In [52]:
#Adding the output layer
model.add(Dense(units = 2, kernel_initializer = 'glorot_uniform', activation = 'softmax'))

In [53]:
model.summary()

Model: "sequential_6"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
dense_19 (Dense)             (None, 10)                20        
_________________________________________________________________
dense_20 (Dense)             (None, 20)                220       
_________________________________________________________________
dense_21 (Dense)             (None, 15)                315       
_________________________________________________________________
dense_22 (Dense)             (None, 2)                 32        
Total params: 587
Trainable params: 587
Non-trainable params: 0
_________________________________________________________________


In [54]:
#Compiling the ANN
model.compile(Adam(lr=0.001), loss = 'sparse_categorical_crossentropy', metrics = ['accuracy'])

In [58]:
model.fit(train_sample,train_label,batch_size=10,nb_epoch=10)

C:\Users\Dell\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: UserWarning: The `nb_epoch` argument in `fit` has been renamed `epochs`.
  """Entry point for launching an IPython kernel.


Epoch 1/10
2000/2000 [==============================] - 2s 1ms/step - loss: 1.3005 - accuracy: 0.5455
Epoch 2/10
2000/2000 [==============================] - 0s 194us/step - loss: 0.5683 - accuracy: 0.6875
Epoch 3/10
2000/2000 [==============================] - 0s 194us/step - loss: 0.5047 - accuracy: 0.7670
Epoch 4/10
2000/2000 [==============================] - 0s 231us/step - loss: 0.4409 - accuracy: 0.8265
Epoch 5/10
2000/2000 [==============================] - 0s 169us/step - loss: 0.4061 - accuracy: 0.8470
Epoch 6/10
2000/2000 [==============================] - 0s 162us/step - loss: 0.3493 - accuracy: 0.8815
Epoch 7/10
2000/2000 [==============================] - 0s 170us/step - loss: 0.3361 - accuracy: 0.8705
Epoch 8/10
2000/2000 [==============================] - 0s 155us/step - loss: 0.2923 - accuracy: 0.8965
Epoch 9/10
2000/2000 [==============================] - 0s 172us/step - loss: 0.2661 - accuracy: 0.9120
Epoch 10/10
2000/2000 [==============================] - 0s 150us/

In [59]:
test_sample=[]
test_label=[]
for i in range(1000):
    younger_ages=randint(13,64)
    test_sample.append(younger_ages)
    test_label.append(0)
    
    older_ages=randint(65,100)
    test_sample.append(older_ages)
    test_label.append(1)

In [60]:
test_sample=np.array(test_sample)
test_label=np.array(test_label)

In [61]:
test_sample

array([63, 89, 38, ..., 85, 54, 75])

In [62]:
test_label

array([0, 1, 0, ..., 1, 0, 1])

In [64]:
test_pred=model.predict_classes(test_sample,batch_size=10)
test_pred

array([1, 1, 0, ..., 1, 1, 1], dtype=int64)

In [67]:
import pandas as pd
compare=pd.DataFrame({'Actual':test_label, 'predicted':test_pred})
compare

Actual  predicted
0          0          1
1          1          1
2          0          0
3          1          1
4          0          0
...      ...        ...
1995       1          1
1996       0          0
1997       1          1
1998       0          1
1999       1          1

[2000 rows x 2 columns]

In [68]:
from sklearn.metrics import confusion_matrix
print(confusion_matrix(test_label,test_pred))

[[ 686  314]
 [   0 1000]]


in this model our predictions are-- correct predictions:1686, wrong predictions:314

In [69]:
from sklearn.metrics import classification_report
print(classification_report(test_label, test_pred))

              precision    recall  f1-score   support

           0       1.00      0.69      0.81      1000
           1       0.76      1.00      0.86      1000

    accuracy                           0.84      2000
   macro avg       0.88      0.84      0.84      2000
weighted avg       0.88      0.84      0.84      2000



In [72]:
from sklearn.metrics import accuracy_score
print(accuracy_score(test_label,test_pred))

0.843
